src/pyaro/timeseries/Filter.py

```diff
@@ -855,4 +859,128 @@ def filter_stations(self, stations: dict[str, Station]) -> dict[str, Station]:
         if self._max_altitude is not None:
             stations = {n: s for n, s in stations.items() if (not math.isnan(s["altitude"]) and s["altitude"] <= self._max_altitude) }
         
-        return stations
\ No newline at end of file
+        return stations
+    
+@registered_filter
+class RelativeAltitudeFilter(StationFilter):
+    """
+    Filter class which filters stations based on the relative difference between
+    the station altitude, and the gridded topography altitude.
+
+    https://github.com/metno/pyaro/issues/39
+    """
+    UNITS_METER = Unit("m")
+    # https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#latitude-coordinate
+    UNITS_LAT = set(["degrees_north", "degree_north", "degree_N", "degrees_N", "degreeN", "degreesN"])
+
+    # https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#longitude-coordinate
+    UNITS_LON = set(["degrees_east", "degree_east", "degree_E", "degrees_E", "degreeE", "degreesE"])
+
+    def __init__(self, topo_file: str | None = None, topo_var: str = "topography", rdiff: float = 0):
+        """
+        :param topo_file : A .nc file from which to read gridded topography data.
+        :param topo_var : Name of variable that stores altitude.
+        :param rdiff : Relative difference (in meters).
+
+        Note:
+        -----
+        - Stations will be kept if abs(altobs-altmod) <= rdiff.
+        - Stations will not be kept if station altitude is NaN.
```

Comment:
What happens with stations outside of the topography-grid? Ideally, log a warning and remove.